### ISSUE TITLE
/run/usr/1001 running out of space due to growing metro-cache

### ISSUE THREAD

**Issue by jakubgs** [status-im/status-react#8597]
Recently builds have been failing due to errors like this:
```
error An unexpected error occurred: "SOME/PATH: ENOSPC: no space left on device, write".
```
But it's not an issue of lack of disk space, but rather lack of space on a virtual filesystem mounted at `/run/usr/1001`, where `1001` is the UID of `jenkins` user on CI hosts.

**Comment by jakubgs** [comment]
At first I thought this was due to something that Nix does, but then after clearing the `metro-cache` folders on all hosts and running the `nightly` build I realized that this happens only for Linux and Windows builds.

**Comment by jakubgs** [comment]
I also found this issue: https://github.com/facebook/metro/issues/331
Which indicated to me that `metro-cache` normally resides at `/tmp/metro-cache`, not in `/usr/run/*`.

**Comment by jakubgs** [comment]
Then I realized that mobile and desktop builds use different `metro` versions:
https://github.com/status-im/status-react/blob/4d8891e35f05e759b82e6e0fd5c5792ad941a387/mobile_files/yarn.lock#L1044
https://github.com/status-im/status-react/blob/4d8891e35f05e759b82e6e0fd5c5792ad941a387/desktop_files/yarn.lock#L5524

Now, those are NOT that far apart, roughly 2:
* [`v0.51.0`](https://github.com/facebook/metro/releases/tag/v0.51.0) - 2018/12/17
* [`v0.48.1`](https://github.com/facebook/metro/releases/tag/v0.48.1) - 2018/09/25

**Comment by jakubgs** [comment]
I attempted to upgrade `metro` for desktop to `0.51.0` but apparently there are some patches applied:
https://github.com/status-im/status-react/blob/4d8891e35f05e759b82e6e0fd5c5792ad941a387/patches/metro%2B0.48.5.patch#L1-L11
and:
https://github.com/status-im/status-react/blob/4d8891e35f05e759b82e6e0fd5c5792ad941a387/patches/metro-config%2B0.48.5.patch#L1-L12

Apparently those were applied by @vkjr in 75af3883b816abe9b3dffe7b92c8d393119e1c86.

Not sure about the second one, but it appears the first one would not be necessary for `0.51.0`:
https://github.com/facebook/metro/issues/146

Not sure though.

**Comment by jakubgs** [comment]
I assume the lack of in my branch(`upgrade-desktop-metro`) `patches/metro-config+0.48.5.patch` is the cause of this error:
```log
Generating /home/jenkins/workspace/tests/linux-build-branch-test/StatusImPackage/index.desktop.bundle and assets folder...

jest-haste-map: @providesModule naming collision:
  Duplicate module name: NetInfo
  Paths: /home/jenkins/workspace/tests/linux-build-branch-test/node_modules/react-native/Libraries/Network/NetInfo.desktop.js collides with /home/jenkins/workspace/tests/linux-build-branch-test/node_modules/react-native/Libraries/Network/NetInfo.js

This error is caused by a @providesModule declaration with the same name across two different files.

Error: jest-haste-map: @providesModule naming collision:
  Duplicate module name: NetInfo
  Paths: /home/jenkins/workspace/tests/linux-build-branch-test/node_modules/react-native/Libraries/Network/NetInfo.desktop.js collides with /home/jenkins/workspace/tests/linux-build-branch-test/node_modules/react-native/Libraries/Network/NetInfo.js

This error is caused by a @providesModule declaration with the same name across two different files.
    at setModule (/home/jenkins/workspace/tests/linux-build-branch-test/node_modules/jest-haste-map/build/index.js:462:17)
    at workerReply (/home/jenkins/workspace/tests/linux-build-branch-test/node_modules/jest-haste-map/build/index.js:512:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
https://ci.status.im/job/tests/job/linux-build-branch-test/1/console

Specifically the part about:
```
Duplicate module name: NetInfo
Paths:
  ${WORK}/node_modules/react-native/Libraries/Network/NetInfo.desktop.js
collides with
  ${WORK}/node_modules/react-native/Libraries/Network/NetInfo.js
```

**Comment by jakubgs** [comment]
I attempted to update the patch for `0.51.1` but that didn't fix the issue:
```patch
patch-package
--- a/node_modules/metro-config/src/defaults/defaults.js
+++ b/node_modules/metro-config/src/defaults/defaults.js
@@ -41,7 +41,7 @@ exports.assetExts = [
 ];
 exports.sourceExts = ["js", "json", "ts", "tsx"];
 exports.moduleSystem = require.resolve("metro/src/lib/polyfills/require.js");
-exports.platforms = ["ios", "android", "windows", "web"];
+exports.platforms = ["ios", "android", "windows", "web", "desktop"];
 exports.providesModuleNodeModules = ["react-native", "react-native-windows"];
 exports.DEFAULT_METRO_MINIFIER_PATH = "metro-minify-uglify";
 exports.defaultCreateModuleIdFactory = defaultCreateModuleIdFactory;
```

**Comment by jakubgs** [comment]
I mean, I could just add an `rm -fr /run/user/${UID}/metro-cache` to the `Jenkinsfile`s and call it a day, but that's hardly a solution, just a hack.

**Comment by jakubgs** [comment]
As far as I can tell it is this stage that actually generates the `/run/user/${UID}/metro-cache` directory:
https://github.com/status-im/status-react/blob/d8645e285929a5d6b18c6e39e0279c590542e83e/scripts/build-desktop.sh#L131-L144
Specifically:
https://github.com/status-im/status-react/blob/d8645e285929a5d6b18c6e39e0279c590542e83e/scripts/build-desktop.sh#L140-L141
Maybe there are some `react-native` flags I could use to change the cache path.

**Comment by jakubgs** [comment]
According to this:
https://gist.github.com/jarretmoses/c2e4786fd342b3444f3bc6beff32098d
This command should clear the cache:
```bash
react-native start --reset-cache
```
But my tests do not show that to be correct.

**Comment by jakubgs** [comment]
Adding `--reset-cache` to `eact-native bundle` in [`scripts/build-desktop.sh`](/scripts/build-desktop.sh) might help, lets see.
https://github.com/status-im/status-react/blob/d8645e285929a5d6b18c6e39e0279c590542e83e/scripts/build-desktop.sh#L140-L141

**Comment by jakubgs** [comment]
I've merged #8621 in the morning and after a while day of builds running the `metro-cache` folders are all the same size:
```
 $ ansible ci-slave-linux -m shell -o -a 'du -hsc /run/user/1001/metro-cache | head -n1'
linux-05.do-ams3.ci.misc | CHANGED | rc=0 | (stdout) 56M      /run/user/1001/metro-cache
linux-04.do-ams3.ci.misc | CHANGED | rc=0 | (stdout) 56M      /run/user/1001/metro-cache
linux-01.do-ams3.ci.misc | CHANGED | rc=0 | (stdout) 56M      /run/user/1001/metro-cache
linux-02.do-ams3.ci.misc | CHANGED | rc=0 | (stdout) 56M      /run/user/1001/metro-cache
linux-03.do-ams3.ci.misc | CHANGED | rc=0 | (stdout) 56M      /run/user/1001/metro-cache
```
Which indicates to me that the addition of `--reset-cache` has worked, and this issue is resolved.

**Comment by jakubgs** [comment]
Though I still would like to upgrade `metro` to the same version as for Mobile...
